lib/network_tools.cpp

```diff
@@ -170,42 +170,27 @@ Matrix addSplitterToFlow(Matrix flow, const Wiring splitter_inputs,
   const int num_splitter_outputs = splitter_outputs.size();
   const int num_splitter_inputs = splitter_inputs.size();
 
-  Row splitter_flow = zeroRow(num_flow_inputs);
-  for (int i = 0; i < num_splitter_inputs; ++i) {
-    splitter_flow.push_back(1.0 / num_splitter_outputs);
-  }
+  const Wiring wired_inputs = wiredConnections(splitter_inputs);
+  const Wiring wired_outputs = wiredConnections(splitter_outputs);
+
+  const double output_fraction = 1.0 / num_splitter_outputs;
 
   // Find this splitter's output in terms of its input flows
-  for (int i = 0; i < num_splitter_inputs; ++i) {
-    if (splitter_inputs[i] != -1) {
-      // Flow to splitter along its i'th input
-      Row input_flow = flow[splitter_inputs[i]];
-      for (int j = 0; j < num_flow_inputs; ++j) {
-        splitter_flow[j] += input_flow[j] / num_splitter_outputs;
-      }
-    }
+  Row input_flow = zeroRow(num_flow_inputs);
```

Comment:
Consider renaming this to something that mentions its relationship to the splitter, like flowToSplitterFromInputs.